### DISCUSSION TITLE
build for foo does not contain a main function

### DISCUSSION THREAD

**Discussion by Funcan** [goreleaser/goreleaser#1700] [Q&A] [answered]
Hi

Probably me misunderstanding something, but there are a bunch of projects I'd like to use goreleaser on, but I run into the same issue on all of them. I created a hello-world repo to demonstrate the issue:

https://github.com/Funcan/goreleasetest

```
$ goreleaser build
...
   ⨯ build failed after 0.19s error=build for goreleasetest does not contain a main function
```

I'm assuming there's something simple but fundamental I'm missing, and would greatly appreciate a hint.

Thanks

**Comment by caarlos0** [comment, marked as answer]
you need to point to the main file/folder in the goreleaser config

check https://goreleaser.com/customization/build/ for details

**Reply by jonhadfield** [reply]
Please could you let me know how this is done using a GitHub Action?

**Comment by Funcan** [comment]
Sorted, thanks!
